test/object-store/c_api/c_api.cpp

```diff
@@ -449,6 +449,99 @@ TEST_CASE("C API (non-database)", "[c_api]") {
             CHECK(std::string{realm_config_get_fifo_path(config.get())} == "test_path.FIFO");
         }
     }
+
+#if REALM_ENABLE_SYNC
+    SECTION("realm_app_config_t") {
+        // realm_http_transport is not defined if REALM_ENABLE_SYNC is not set
+        auto http_transport = realm_http_transport(nullptr);
+        auto app_config = cptr(realm_app_config_new("some_app_id", &http_transport));
+
+#if REALM_ENABLE_AUTH_TESTS
```

Comment:
Is there value to testing this with sync enabled but without a server to talk to?

Reply:
Cleaned up and put single test under `#if REALM_ENABLE_AUTH_TESTS`